### PR TITLE
feat: increase stake of a sns neuron

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Add a feature to increase the stake of a Sns neuron.

# ⚠️ Warning

As for "staking Sns neuron", this PR follows the existing same Sns pattern that propose a payment flow which happens on the frontend side only. Therefore, it isn't fully reliable. e.g. if transfer succeed and refresh isn't called, the stake of the neuron is not increase while the token haven been transfered. This would need refreshing the neuron again, the user would have to increase the stake again. The resulting would be the sum of both actions. I decline any responsability about this.

Worth to note that we will soon begin to strenghtness the payment flows therefore this will be ultimately handled in a better way.

# PRs

- [x] need sns-js https://github.com/dfinity/ic-js/pull/248 

# Changes

- implement service and api to increase stake of a sns neuron
- add a button to the sns neuron detail page to open the modal to process the increase of the stake (similar to nns)
- implementation transaction modal flow and call increase stake neuron on execution (similar flow to nns)
- load params and accounts - `snsSyncAccounts` - if necessay on button click and display a spinner while loading the data
- misc. files name refactoring to make obvious which component is Nns and which is Sns

### CODE REVIEW

**Review by mstrasinskis — APPROVED** [review]
LGTM, thanks, just a couple of question.

File: frontend/src/lib/modals/sns/neurons/SnsIncreaseStakeNeuronModal.svelte [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    on:nnsClose
+    bind:currentStep
+    {token}
+    {transactionFee}
+    {governanceCanisterId}
+  >
+    <svelte:fragment slot="title"
+      >{title ?? $i18n.accounts.new_transaction}</svelte:fragment
+    >
+    <p slot="description" class="value">here description</p>
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/src/lib/modals/sns/neurons/SnsIncreaseStakeNeuronModal.svelte]: 🤦‍♂️. Done.

File: frontend/src/lib/components/sns-neuron-detail/SnsNeuronInfoStake.svelte [hunk 2]
```diff
@@ -64,6 +68,9 @@
     {#if allowedToDissolve && ENABLE_SNS}
       <IncreaseSnsDissolveDelayButton />
     {/if}
+    {#if ENABLE_SNS_2}
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/src/lib/components/sns-neuron-detail/SnsNeuronInfoStake.svelte:71]: No idea 🤷‍♂️.

File: frontend/src/lib/modals/neurons/DisburseSnsNeuronModal.svelte [hunk 3]
```diff
@@ -70,7 +70,7 @@
       }
 
       loading = true;
-      await syncSnsAccounts(selectedProjectCanisterId);
+      await syncSnsAccounts({ rootCanisterId: selectedProjectCanisterId });
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/src/lib/modals/neurons/DisburseSnsNeuronModal.svelte:73]: Discussed in a chat, we clear that later and if needed, we refactor in another PR.

File: frontend/src/lib/modals/sns/neurons/SnsIncreaseStakeNeuronModal.svelte [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    startBusy({
+      initiator: "load-sns-accounts",
+    });
+
+    await syncSnsAccounts({
+      rootCanisterId,
+      handleError: () => stopBusySpinner(),
+    });
+  });
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/src/lib/modals/sns/neurons/SnsIncreaseStakeNeuronModal.svelte:47]: It is few lines later (line 93)

```
$: loading,
    (() => {
      if (loading) {
        return;
      }

      stopBusySpinner();
    })();
```

**Pull request merged**